base/file.jl

```diff
@@ -330,6 +324,19 @@ function rm(path::AbstractString; force::Bool=false, recursive::Bool=false, allo
 end
 
 
+# Loaded DLLs cannot be deleted on Windows, even with posix delete mode but they can be renamed.
+# delayed_delete_dll(path) does so temporarily, until later cleanup by Pkg.gc().
+function delayed_delete_dll(path)
+    # in-use DLL must be kept on the same drive
+    temp_path = tempname(abspath(dirname(path)); cleanup=false, suffix=string("_", basename(path)))
+    @debug "Could not delete DLL most likely because it is loaded, moving to a temporary path" path temp_path
+    mkpath(delayed_delete_ref())
+    io = last(mktemp(delayed_delete_ref(); cleanup=false))
+    print(io, temp_path) # record the temporary path for Pkg.gc()
+    close(io)
```

Comment:
I suggest using `open() do io` form here, just in case the `print()` throws an error (out of space?) which would then cause us to not call `close(io)` until we eventually GC.  Considering we could end up attempting to delete many files in a row and hit the same error over and over again, it'd be nice to ensure that we're closing `io` immediately so we don't accidentally have many file handles stacking up.

I think it is clever to name the temp path ending in `.dll`, this ensures that future attempts to delete this file (if we fail to record the reference here) will go through the same code path, so it has a hope to be cleaned up in the future.